File: src/main/java/WrightGL/Geometry.java
package WrightGL;

public class Geometry {
    public static float[] triangle = {
            0,0,0,
            1,1,0,
            0,1,0,
            0,0,0,
            1,0,0,
            1,1,0,
    };
    public static float triangleStripQuad[] = { -1, -1, 1, -1, -1, 1, 1, 1 };
    public static float triangleNormals[] = {
            -1, -1,0, 0, 0, -1,
            1, -1, 0, 0, 0, -1,
            0,  1, 0, 0, 0, -1
    };
    public static float cubeNormals[] = {
            // left
            -1.0f,-1.0f,-1.0f,        -1, 0, 0,
            -1.0f,-1.0f, 1.0f,        -1, 0, 0,
            -1.0f, 1.0f, 1.0f,         -1, 0, 0,

            -1.0f,-1.0f,-1.0f,        -1, 0, 0,
            -1.0f, 1.0f, 1.0f,        -1, 0, 0,
            -1.0f, 1.0f,-1.0f,        -1, 0, 0,

            // down
            1.0f,-1.0f, 1.0f,        0, -1, 0,
            -1.0f,-1.0f, 1.0f,        0, -1, 0,
            -1.0f,-1.0f,-1.0f,        0, -1, 0,

            1.0f,-1.0f, 1.0f,        0, -1, 0,
            -1.0f,-1.0f,-1.0f,        0, -1, 0,
            1.0f,-1.0f,-1.0f,        0, -1, 0,

            // back
            1.0f, 1.0f,-1.0f,          0, 0,-1,
            -1.0f,-1.0f,-1.0f,         0, 0,-1,
            -1.0f, 1.0f,-1.0f,         0, 0,-1,

            1.0f, 1.0f,-1.0f,         0, 0,-1,
            1.0f,-1.0f,-1.0f,         0, 0,-1,
            -1.0f,-1.0f,-1.0f,         0, 0,-1,

            // front
            -1.0f, 1.0f, 1.0f,       0, 0, 1,
            -1.0f,-1.0f, 1.0f,       0, 0, 1,
            1.0f,-1.0f, 1.0f,        0, 0, 1,

            1.0f, 1.0f, 1.0f,        0, 0, 1,
            -1.0f, 1.0f, 1.0f,       0, 0, 1,
            1.0f,-1.0f, 1.0f,        0, 0, 1,

            // right
            1.0f, 1.0f, 1.0f,        1, 0, 0,
            1.0f,-1.0f,-1.0f,        1, 0, 0,
            1.0f, 1.0f,-1.0f,        1, 0, 0,

            1.0f,-1.0f,-1.0f,        1, 0, 0,
            1.0f, 1.0f, 1.0f,        1, 0, 0,
            1.0f,-1.0f, 1.0f,        1, 0, 0,

            // up
            1.0f, 1.0f, 1.0f,        0, 1, 0,
            1.0f, 1.0f,-1.0f,        0, 1, 0,
            -1.0f, 1.0f,-1.0f,       0, 1, 0,

            1.0f, 1.0f, 1.0f,        0, 1, 0,
            -1.0f, 1.0f,-1.0f,       0, 1, 0,
            -1.0f, 1.0f, 1.0f,       0, 1, 0

    };
}

File: src/main/java/_0_1/math/Geometry.kt
package _0_1.math

object Geometry {
    var triangle = floatArrayOf(0f, 0f, 0f, 1f, 1f, 0f, 0f, 1f, 0f, 0f, 0f, 0f, 1f, 0f, 0f, 1f, 1f, 0f)
    var quadTriangleStrip = floatArrayOf(-1f, -1f, 1f, -1f, -1f, 1f, 1f, 1f)
    var triangleNormals = floatArrayOf(-1f, -1f, 0f, 0f, 0f, -1f, 1f, -1f, 0f, 0f, 0f, -1f, 0f, 1f, 0f, 0f, 0f, -1f)
    var cubeNormals = floatArrayOf( // left
        -1.0f, -1.0f, -1.0f, -1f, 0f, 0f,
        -1.0f, -1.0f, 1.0f, -1f, 0f, 0f,
        -1.0f, 1.0f, 1.0f, -1f, 0f, 0f,
        -1.0f, -1.0f, -1.0f, -1f, 0f, 0f,
        -1.0f, 1.0f, 1.0f, -1f, 0f, 0f,
        -1.0f, 1.0f, -1.0f, -1f, 0f, 0f,  // down
        1.0f, -1.0f, 1.0f, 0f, -1f, 0f,
        -1.0f, -1.0f, 1.0f, 0f, -1f, 0f,
        -1.0f, -1.0f, -1.0f, 0f, -1f, 0f,
        1.0f, -1.0f, 1.0f, 0f, -1f, 0f,
        -1.0f, -1.0f, -1.0f, 0f, -1f, 0f,
        1.0f, -1.0f, -1.0f, 0f, -1f, 0f,  // back
        1.0f, 1.0f, -1.0f, 0f, 0f, -1f,
        -1.0f, -1.0f, -1.0f, 0f, 0f, -1f,
        -1.0f, 1.0f, -1.0f, 0f, 0f, -1f,
        1.0f, 1.0f, -1.0f, 0f, 0f, -1f,
        1.0f, -1.0f, -1.0f, 0f, 0f, -1f,
        -1.0f, -1.0f, -1.0f, 0f, 0f, -1f,  // front
        -1.0f, 1.0f, 1.0f, 0f, 0f, 1f,
        -1.0f, -1.0f, 1.0f, 0f, 0f, 1f,
        1.0f, -1.0f, 1.0f, 0f, 0f, 1f,
        1.0f, 1.0f, 1.0f, 0f, 0f, 1f,
        -1.0f, 1.0f, 1.0f, 0f, 0f, 1f,
        1.0f, -1.0f, 1.0f, 0f, 0f, 1f,  // right
        1.0f, 1.0f, 1.0f, 1f, 0f, 0f,
        1.0f, -1.0f, -1.0f, 1f, 0f, 0f,
        1.0f, 1.0f, -1.0f, 1f, 0f, 0f,
        1.0f, -1.0f, -1.0f, 1f, 0f, 0f,
        1.0f, 1.0f, 1.0f, 1f, 0f, 0f,
        1.0f, -1.0f, 1.0f, 1f, 0f, 0f,  // up
        1.0f, 1.0f, 1.0f, 0f, 1f, 0f,
        1.0f, 1.0f, -1.0f, 0f, 1f, 0f,
        -1.0f, 1.0f, -1.0f, 0f, 1f, 0f,
        1.0f, 1.0f, 1.0f, 0f, 1f, 0f,
        -1.0f, 1.0f, -1.0f, 0f, 1f, 0f,
        -1.0f, 1.0f, 1.0f, 0f, 1f, 0f
    )
}